### PR TITLE
Swap update

### PR DESCRIPTION
**Pull request by letier3110**
https://www.notion.so/madfissolutions/Redesign-swap-section-need-to-include-smart-router-functionality-bd8d49a2c1594b2087fa182a2680881d

### CODE REVIEW

File: metro.config.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ module.exports = (async () => {
     resolver: {
       extraNodeModules: require('node-libs-react-native'),
       assetExts: assetExts.filter(ext => ext !== 'svg'),
-      sourceExts: [...sourceExts, 'svg']
+      sourceExts: [...sourceExts, 'jsx', 'svg']
```
Comment (unixvb) [inline metro.config.js:22]: why?

File: src/hooks/slippage-tolerance/slippage-tolerance.provider.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React, { FC, useState } from 'react';
+
+import { SlippageToleranceContext } from './slippage-tolerance.context';
+
+export const SlippageToleranceProvider: FC = ({ children }) => {
+  const [slippageTolerance, setSlippageTolerance] = useState(0);
```
Comment (unixvb) [inline src/hooks/slippage-tolerance/slippage-tolerance.provider.tsx:6]: save `slippageTolerance` into redux state

File: src/components/asset-amount-input/asset-amount-input.props.ts [hunk 3]
```diff
@@ -13,4 +13,5 @@ export interface AssetAmountInputProps {
   onBlur?: EmptyFn;
   onFocus?: TextInputProps['onFocus'];
   onValueChange: EventFn<AssetAmountInterface>;
+  isEditable?: boolean;
```
Comment (unixvb) [inline src/components/asset-amount-input/asset-amount-input.props.ts:16]: `disabled`

File: src/components/asset-amount-input/asset-amount-input.tsx [hunk 4]
```diff
@@ -48,7 +48,9 @@ export const AssetAmountInput: FC<AssetAmountInputProps> = ({
   isError = false,
   onBlur,
   onFocus,
-  onValueChange
+  onValueChange,
+  isEditable = true
+  // showExchangeRate = true
```
Comment (unixvb) [inline src/components/asset-amount-input/asset-amount-input.tsx:53]: delete commented code

File: src/interfaces/swap-asset.interface.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Trade } from 'swap-router-sdk';
+
+import { AssetAmountInterface } from '../components/asset-amount-input/asset-amount-input';
+
+export interface SwapFormValues {
+  inputAssets: AssetAmountInterface;
+  outputAssets: AssetAmountInterface;
+  bestTrade: Trade | [];
+  bestTradeWithSlippageTolerance: Trade | [];
```
Comment (unixvb) [inline src/interfaces/swap-asset.interface.ts:9]: do you really need to save `bestTrade` and `bestTradeWithSlippageTolerance` inside form?  user do not input them

Comment (unixvb) [inline src/interfaces/swap-asset.interface.ts:9]: `Trade | []` > `Trade`

File: src/screens/swap/swap-router.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <>
+      <View style={styles.swapInfoContainer}>
+        {trade.map((item, index) => {
+          return (
+            <SwapRouterItem
+              key={`${index}_${item.dexType}_${item.aTokenSlug}_${item.bTokenSlug}`}
+              tradeOperation={item}
+              // isShowNextArrow={index !== trade.length - 1}
```
Comment (unixvb) [inline src/screens/swap/swap-router.tsx:27]: ?

File: src/screens/swap/swap.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    };
+
+    trackEvent('SWAP_FORM_SUBMIT', AnalyticsEventCategory.FormSubmit, analyticsProperties);
+    const routingFeeOpParams = await getRoutingFeeTransferParams(
+      inputMutezAmount,
+      bestTradeWithSlippageTolerance,
+      selectedAccount.publicKeyHash,
+      tezos
+    );
+    // @ts-ignore
```
Comment (unixvb) [inline src/screens/swap/swap.tsx:66]: why?

File: src/interfaces/swap-asset.interface.ts (hunk 5 above)
Comment (letier3110) [reply src/interfaces/swap-asset.interface.ts:9]: yes, because form and submit handler in different component

**Review by unixvb — APPROVED** [review]

File: src/components/swap-price-update-bar/swap-price-update-bar.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
+import React, { FC, useEffect, useMemo, useRef, useState } from 'react';
+import { Animated, Text, View } from 'react-native';
+import { useAllRoutePairs } from 'swap-router-sdk';
+
+import { formatSize } from '../../styles/format-size';
+import { Divider } from '../divider/divider';
+import { useSwapPriceUpdateBarStyles } from './swap-price-update-bar.styles';
+
+export const BLOCK_DURATION = 30000;
+export const TEZOS_DEXES_API_URL = 'wss://tezos-dexes-api-mainnet.production.madservice.xyz';
```
Comment (LendiHop) [inline src/components/swap-price-update-bar/swap-price-update-bar.tsx:10]: this is already exported from swap-form-tsx

File: src/screens/swap/swap-form.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+import { useSlippageSelector } from '../../store/settings/settings-selectors';
+import { useTezosTokenSelector, useVisibleAssetListSelector } from '../../store/wallet/wallet-selectors';
+import { formatSize } from '../../styles/format-size';
+import { emptyToken, TokenInterface } from '../../token/interfaces/token.interface';
+import { getTokenSlug } from '../../token/utils/token.utils';
+import { SwapExchangeRate } from './swap-exchange-rate';
+import { SwapRoute } from './swap-router';
+import { useSwapStyles } from './swap.styles';
+
+export const TEZOS_DEXES_API_URL = 'wss://tezos-dexes-api-mainnet.production.madservice.xyz';
```
Comment (LendiHop) [inline src/screens/swap/swap-form.tsx:32]: here

**Review by LendiHop — APPROVED** [review]

**Pull request merged**